plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -446,6 +503,10 @@ else if (partitionKeys.containsKey(column.getId())) {
                 else if (column.isPathColumn()) {
                     columnAdaptations.add(ColumnAdaptation.constantColumn(nativeValueToBlock(FILE_PATH.getType(), utf8Slice(path.toString()))));
                 }
+                else if (column.isUpdateRowIdColumn()) {
+                    // $row_id is a composite of multiple physical columns, it is assembled by the IcebergPageSource
```

Comment:
The comma usage here is bad grammar. Should be a semicolon or separate sentences.